tests/layer1/test_port_error.py

```diff
@@ -44,23 +45,29 @@ def get_mac_fault_count(dut, interface, fault_type):
     def get_interface_status(dut, interface):
         return dut.show_and_parse("show interfaces status {}".format(interface))[0].get("oper", "unknown")
 
-    @pytest.fixture
+    @pytest.fixture(scope="class")
     def select_random_interfaces(self, duthosts, enum_rand_one_per_hwsku_frontend_hostname):
         dut = duthosts[enum_rand_one_per_hwsku_frontend_hostname]
-        interfaces = list(dut.show_and_parse("show interfaces status"))
 
         sfp_presence = dut.command(cmd_sfp_presence)
         parsed_presence = {line.split()[0]: line.split()[1] for line in sfp_presence["stdout_lines"][2:]}
 
-        available_interfaces = [
-            intf["interface"] for intf in interfaces
-            if parsed_presence.get(intf["interface"]) == "Present"
-        ]
+        eeprom_infos = dut.command("sudo sfputil show eeprom -d")['stdout']
+        eeprom_infos = parse_sfp_eeprom_infos(eeprom_infos)
+
+        available_interfaces = []
```

Comment:
we should define a better name for this list, perhaps optical_interfaces

Reply:
changed to "available_optical_interfaces"